Extension/src/LanguageServer/lmTool.ts

```diff
@@ -0,0 +1,109 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as vscode from 'vscode';
+import { localize } from 'vscode-nls';
+import * as util from '../common';
+import * as logger from '../logger';
+import * as telemetry from '../telemetry';
+import { ChatContextResult } from './client';
+import { getClients } from './extension';
+
+const knownValues: { [Property in keyof ChatContextResult]?: { [id: string]: string } } = {
+    language: {
+        'c': 'C',
+        'cpp': 'C++',
+        'cuda-cpp': 'CUDA C++'
+    },
+    compiler: {
+        'msvc': 'MSVC',
+        'clang': 'Clang',
+        'gcc': 'GCC'
+    },
+    standardVersion: {
+        'c++98': 'C++98',
+        'c++03': 'C++03',
+        'c++11': 'C++11',
+        'c++14': 'C++14',
+        'c++17': 'C++17',
+        'c++20': 'C++20',
+        'c++23': 'C++23',
+        'c90': "C90",
+        'c99': "C99",
+        'c11': "C11",
+        'c17': "C17",
+        'c23': "C23"
+    },
+    targetPlatform: {
+        'windows': 'Windows',
+        'Linux': 'Linux',
+        'macos': 'macOS'
+    }
+};
+
+class StringLanguageModelToolResult implements vscode.LanguageModelToolResult

```

Comment:
Can you run the formatter on this document?

Reply:
![image](https://github.com/user-attachments/assets/c54ebe92-209d-4b35-baf8-eb227806d0c6)

Reply:
yes, done